SourceCode/Ray.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Ray.h"
+
+#include <cmath>
+
+Vector::Vector() : x(0), y(0), z(0){};
+
+Vector::Vector(float x, float y, float z) : x(x), y(y), z(z){};
+
+Vector Vector::operator-(const Vector& other) const {
+  return {this->x - other.x, this->y - other.y, this->z - other.z};
+}
+
+Vector Vector::operator+(const Vector& other) const {
+  return {this->x + other.x, this->y + other.y, this->z + other.z};
+}
+
+void Vector::normalize() {
+  float length = this->length();
+  this->x = this->x / this->length();
```

Comment:
You're caching the length, but do not use it afterwards, also you could go with only one division